tests/Feature/BookTest.php

```diff
@@ -249,4 +249,44 @@ public function test_getBookGenre_success(): void
                     });
             });
     }
+
+    public function test_addBook_success(): void
+    {
+        Genre::factory()->create();
+
+        $testData = [
+            'title' => 'A boooooook',
+            'author' => 'An author',
+            'genre_id' => 1
+        ];
+
+        $response = $this->postJson('/api/books', $testData);
+
+        $response->assertStatus(201)
+        ->assertJson(function (AssertableJson $json) {
+            $json->hasAll(['message', 'success']);
+        });
+
+        $this->assertDatabaseHas('books', $testData );
+    }
+
+    public function test_addBook_failure(): void
+    {
+        $testData = [
+            'title' => 'A boooooook'
+        ];
+
+        $response = $this->postJson('/api/books', $testData);
+
+        $response->assertStatus(422)
+        ->assertJson(function(AssertableJson $json) {
+            $json->hasAll(['message', 'errors']);
+        });
```

Comment:
You should use assertInvalid here instead - it'll automatically check for specific errors

Reply:
https://laravel.com/docs/11.x/http-tests#validation-assert-invalid